### PR TITLE
Chassis code

### PR DESCRIPTION
**Pull request by joshuatehrlich**
This is the chassis code, adapted from 2018 code.

### CODE REVIEW

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 1]
```diff
@@ -1,32 +1,108 @@
...
+        Component.leftWheelEncoder.setDistancePerPulse(Metrics.Wheel.INCHES_PER_TICK);
+        Component.rightWheelEncoder.setDistancePerPulse(Metrics.Wheel.INCHES_PER_TICK);
+		Component.leftWheelAccelerationCap = new EnableableModifier(new AccelerationCap(Component.pdp));
+		Component.leftWheelAccelerationCap.enable();
+		Component.rightWheelAccelerationCap = new EnableableModifier(new AccelerationCap(Component.pdp));
+		Component.rightWheelAccelerationCap.enable();
+        Component.leftWheel = new Motor("LeftWheel", Component.leftWheelAccelerationCap,
+			new VictorSP(Port.PWM.leftDriveA), new VictorSP(Port.PWM.leftDriveB));
+		Component.rightWheel = new Motor("RightWheel", Component.rightWheelAccelerationCap,
+            new VictorSP(Port.PWM.rightDriveA), new VictorSP(Port.PWM.rightDriveB));
```
Comment (NikhilSuresh24) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: Make these CANTalonSRX instead of VictorSP

**Review by NikhilSuresh24 — CHANGES_REQUESTED** [review]
Overall looks good. One main change needed for motorcontrollers.

**Comment by countableclouds** [comment]
Just fixed PR changes...

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 2]
```diff
@@ -1,32 +1,123 @@
...
+
+import org.usfirst.frc4904.standard.custom.sensors.PDP;
+import org.usfirst.frc4904.standard.custom.PCMPort;
+import org.usfirst.frc4904.standard.custom.sensors.CANEncoder;
+import org.usfirst.frc4904.standard.custom.sensors.EncoderPair;
+import org.usfirst.frc4904.standard.custom.motioncontrollers.CANTalonSRX;
+import org.usfirst.frc4904.standard.custom.motioncontrollers.CustomPIDController;
+import org.usfirst.frc4904.standard.custom.sensors.NavX;
+
+import edu.wpi.first.wpilibj.VictorSP;
```
Comment (Daniel-E-B) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: We almost definitely will not use victors this year. We will likely use talons (and maybe some sparks, which are now at `org.usfirst.frc4904.standard.custom.motioncontrollers.Spark`).

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 3]
```diff
@@ -1,32 +1,123 @@
...
+            new CANTalonSRX(Port.PWM.rightDriveA), new CANTalonSRX(Port.PWM.rightDriveB));
+		// Chassis
+		Component.shifter = new SolenoidShifters(Port.Pneumatics.shifter.pcmID, Port.Pneumatics.shifter.forward,
+            Port.Pneumatics.shifter.reverse);
+        Component.chassisEncoders = new EncoderPair(Component.leftWheelEncoder, Component.rightWheelEncoder);
+        Component.chassis = new TankDriveShifting("2019-Chassis", Component.leftWheel, Component.rightWheel, Component.shifter);
+        HumanInput.Driver.xbox = new CustomXbox(Port.HumanInput.xboxController);
+        HumanInput.Driver.xbox.setDeadZone(HumanInterfaceConfig.XBOX_DEADZONE);
+        HumanInput.Operator.joystick = new CustomJoystick(Port.HumanInput.joystick);
+        HumanInput.Operator.joystick.setDeadzone(HumanInterfaceConfig.STICK_LEFT_DEADZONE);
```
Comment (Daniel-E-B) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: the stick deadzone is set to`STICK_LEFT_DEADZONE`, and `STICK_RIGHT_DEADZONE` is left unused. What is the purpose of `STICK_RIGHT_DEADZONE` and where should it be used?

**Review by Daniel-E-B — CHANGES_REQUESTED** [review]
I'm glad that `Component.driverXbox` got fixed :). Other than deciding on speed controllers, and either removing or using `STICK_RIGHT_DEADZONE` from `HumanInterfaceConfig`, this looks great!

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java (hunk 3 above)
Comment (countableclouds) [reply src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: Talked to Billy, changing to JOYSTICK_DEADZ
ONE...

**Review by Daniel-E-B — DISMISSED** [review]

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 4]
```diff
@@ -1,32 +1,153 @@
...
 
 public class RobotMap {
     public static class Port {
         public static class HumanInput {
+            public static final int joystick = 0;
             public static final int xboxController = 1;
+        }
+
+        public static class CANMotor {
+            public static final int leftDriveA = 24; //TODO: MIGHT CHANGE
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: Make all of these -1

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 5]
```diff
@@ -1,32 +1,153 @@
...
+            public static final int leftDriveB = 2;
+            public static final int rightDriveA = 3;
+            public static final int rightDriveB = 4;
+        }
+
+        public static class PWM {
```
Comment (billwpierce) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: delete if unused

**Review by billwpierce — COMMENTED** [review]
Looks Good --> I'm gonna to add config files and make a PR

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 6]
```diff
@@ -1,32 +1,153 @@
...
-		Component.driverXbox.setDeadZone(0.1);
+        /* General */
+        Component.pdp = new PDP();
+        Component.navx = new NavX(SerialPort.Port.kMXP);
+        /* Drive Train */
+        // Wheel Encoders
+        Component.leftWheelEncoder = new CANEncoder("LeftEncoder", Port.CAN.leftEncoder);
+        Component.rightWheelEncoder = new CANEncoder("RightEncoder", Port.CAN.rightEncoder);
+        Component.chassisEncoders = new EncoderPair(Component.leftWheelEncoder, Component.rightWheelEncoder);
+        Component.leftWheelEncoder.setDistancePerPulse(Metrics.Wheel.INCHES_PER_TICK);
```
Comment (oshlern) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: Set in the constructor of leftWheelEncoder

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 7]
```diff
@@ -1,32 +1,153 @@
...
+        Component.rightWheelEncoder.setDistancePerPulse(Metrics.Wheel.INCHES_PER_TICK);
```
Comment (oshlern) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: Set in the constructor of rightWheelEncoder

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 8]
```diff
@@ -1,32 +1,153 @@
...
+        Component.rightWheelAccelerationCap = new EnableableModifier(new AccelerationCap(Component.pdp));
+        Component.rightWheelAccelerationCap.enable();
+        // Wheels
+        Component.rightWheelA = new Motor("rightWheelA", true, Component.rightWheelAccelerationCap,
+            new CANTalonSRX(Port.CANMotor.rightDriveA));
+        Component.rightWheelB = new Motor("rightWheelB", Component.rightWheelAccelerationCap,
+            new CANTalonSRX(Port.CANMotor.rightDriveB));
+        Component.leftWheelA = new Motor("leftWheelA", true, Component.leftWheelAccelerationCap,
+            new CANTalonSRX(Port.CANMotor.leftDriveA));
+        Component.leftWheelB = new Motor("leftWheelB", true, Component.leftWheelAccelerationCap,
```
Comment (oshlern) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: Combine A and B by calling the motor constructor with 2 CANTalonSRX inputs

File: src/main/java/org/usfirst/frc4904/robot/RobotMap.java [hunk 9]
```diff
@@ -1,32 +1,153 @@
...
+            new CANTalonSRX(Port.CANMotor.leftDriveB));
+        // Shifter
+        Component.shifter = new SolenoidShifters(Port.Pneumatics.shifter.pcmID, Port.Pneumatics.shifter.forward,
+            Port.Pneumatics.shifter.reverse);
+        // General Chassis
+        Component.chassis = new TankDriveShifting("2019-Chassis", Component.leftWheelA, Component.leftWheelB,
+            Component.rightWheelA, Component.rightWheelB, Component.shifter);
+        Component.drivePID = new CustomPIDController(PID.Drive.P, PID.Drive.I, PID.Drive.D, PID.Drive.F,
```
Comment (oshlern) [inline src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: Use an EncoderPair to combine left and right encoders

Comment (oshlern) [reply src/main/java/org/usfirst/frc4904/robot/RobotMap.java]: we didn't do this last year because one of the encoders was unplugged

**Review by oshlern — DISMISSED** [review]

**Review by NikhilSuresh24 — DISMISSED** [review]

**Review by NikhilSuresh24 — APPROVED** [review]

**Pull request merged**